<div style="font-family: Arial;">
<h3 style="text-align: center;">
<b>ESPECIALIZACIÓN EN CIENCIA DE DATOS E INTELIGENCIA ARTIFICIAL</b>
</h3>
<h3 style="text-align: center; color:  #C2262B;"> <!--Color institucional-->
<strong>Introducción a la Ciencia de Datos</strong>
</h3>
<h3>
<b>Tema:</b> EDA: Rendimiento Académico
</h3>
<hr style="width:100%; border:1px solid white;">
<h6 style="text-align: left; margin-bottom: 5px">
<span style="font-size:18px;">
DOCENTE: Mateo Cano Solis
</span><br>
<span style="font-size:18px;">
INTEGRANTES: Johanna Hincapié, Milena Londoño, Daniela Franco y Julián Trujillo
</span><br>
<span style="font-size:18px;">
FECHA: 16-04-2026
</div>

## **Exploración inicial**

Una institución educativa busca comprender los factores que influyen en el rendimiento académico de los estudiantes. Para ello, se recopiló información demográfica, hábitos de estudio, participación en actividades y apoyo familiar.

El propósito es identificar los factores que impactan el desempeño académico para mejorar estrategias pedagógicas y fortalecer programas de apoyo.

### **Objetivo**

Se requiere analizar cómo factores personales, familiares y académicos influyen en el rendimiento estudiantil, medido mediante el promedio académico y la clasificación del desempeño. 

Este análisis permitirá:

- Identificar variables clave del rendimiento
- Comprender comportamientos académicos
- Apoyar la toma de decisiones educativas


### **Variables observadas**

El conjunto de datos integra información sobre características personales, contexto familiar, hábitos académicos y participación en actividades extracurriculares. Esto permite realizar un análisis integral del rendimiento estudiantil. El dataset contiene la siguiente información:


| Nombre de la variable | Descripción                                    | Tipo                  | Subtipo      | Valores                                                                                                                      | Unidades |
| --------------------- | ---------------------------------------------- | --------------------- | ------------ | ---------------------------------------------------------------------------------------------------------------------------- | -------- |
| StudentID             | Identificador único del estudiante             | Categórica            | Nominal (ID) | Valores numéricos únicos                                                                                                     | -        |
| Age                   | Edad del estudiante                            | Cuantitativa discreta | Discreta     | Continua de 15 a 18 aprox                                                                                                    | años     |
| Gender                | Género del estudiante                          | Categórica binaria    | Binaria      | 0: Masculino<br>1: Femenino                                                                                                  | -        |
| Ethnicity             | Grupo étnico del estudiante                    | Categórica nominal    | Nominal      | 0: Caucasian<br>1: African American<br>2: Asian<br>3: Other                                                                  | -        |
| ParentalEducation     | Nivel educativo de los padres                  | Categórica ordinal    | Ordinal      | 0: None<br>1: High School<br>2: Some College<br>3: Bachelor's<br>4: Higher                                                   | -        |
| StudyTimeWeekly       | Horas de estudio semanal                       | Cuantitativa continua | Continua     | Continua según horas registradas                                                                                             | horas    |
| Absences              | Número de ausencias escolares                  | Cuantitativa discreta | Discreta     | Valores enteros ≥ 0                                                                                                          | días     |
| Tutoring              | Indica si recibe tutorías académicas           | Categórica binaria    | Binaria      | 0: No<br>1: Yes                                                                                                               | -        |
| ParentalSupport       | Nivel de apoyo parental percibido              | Categórica ordinal    | Ordinal      | 0: None<br>1: Low<br>2: Moderate<br>3: High<br>4: Very High                                                                  | -        |
| Extracurricular       | Participación en actividades extracurriculares | Categórica binaria    | Binaria      | 0: No<br>1: Yes                                                                                                               | -        |
| Sports                | Participación en deportes                      | Categórica binaria    | Binaria      | 0: No<br>1: Yes                                                                                                               | -        |
| Music                 | Participación en actividades musicales         | Categórica binaria    | Binaria      | 0: No<br>1: Yes                                                                                                               | -        |
| Volunteering          | Participación en voluntariado                  | Categórica binaria    | Binaria      | 0: No<br>1: Yes                                                                                                               | -        |
| GPA                   | Promedio académico del estudiante              | Cuantitativa continua | Continua     | Continua de 0 a 4 aprox                                                                                                      | puntos   |
| GradeClass            | Clasificación del rendimiento académico        | Categórica ordinal    | Ordinal      | 0: 'A' (GPA ≥ 3.5)<br>1: 'B' (3.0 ≤ GPA < 3.5)<br>2: 'C' (2.5 ≤ GPA < 3.0)<br>3: 'D' (2.0 ≤ GPA < 2.5)<br>4: 'F' (GPA < 2.0) | -        |






## **Importación de librerías necesarias para el análisis estadístico:**

In [3]:
import numpy as np # Manipulación de datos y cálculos numéricos
import pandas as pd # Manipulación de datos en formato tabular (dataframe)
import matplotlib.pyplot as plt # Visualización de datos

## **Revisión del conjunto de datos:**

In [4]:
# Carga de los datos desde un archivo CSV utilizando pandas y visualización de las primeras filas del dataframe para verificar su contenido.

df = pd.read_csv("Student_performance_data.csv")

df.head(10)


,StudentID,Age,Gender,Ethnicity,ParentalEducation,StudyTimeWeekly,Absences,Tutoring,ParentalSupport,Extracurricular,Sports,Music,Volunteering,GPA,GradeClass
0,1001,17,1,0,2,19.833723,7,1,2,0,0,1,0,2.929196,2.0
1,1002,18,0,0,1,15.408756,0,0,1,0,0,0,0,3.042915,1.0
2,1003,15,0,2,3,4.210570,26,0,2,0,0,0,0,0.112602,4.0
3,1004,17,1,0,3,10.028829,14,0,3,1,0,0,0,2.054218,3.0
4,1005,17,1,0,2,4.672495,17,1,3,0,0,0,0,1.288061,4.0
5,1006,18,0,0,1,8.191219,0,0,1,1,0,0,0,3.084184,1.0
6,1007,15,0,1,1,15.601680,10,0,3,0,1,0,0,2.748237,2.0
7,1008,15,1,1,4,15.424496,22,1,1,1,0,0,0,1.360143,4.0
8,1009,17,0,0,0,4.562008,1,0,2,0,1,0,1,2.896819,2.0
9,1010,16,1,0,1,18.444466,0,0,3,1,0,0,0,3.573474,0.0


In [5]:
# Se obtiene un resumen de la estructura del conjunto de datos con el médoto '.info()'

df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2392 entries, 0 to 2391
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   StudentID          2392 non-null   int64  
 1   Age                2392 non-null   int64  
 2   Gender             2392 non-null   int64  
 3   Ethnicity          2392 non-null   int64  
 4   ParentalEducation  2392 non-null   int64  
 5   StudyTimeWeekly    2392 non-null   float64
 6   Absences           2392 non-null   int64  
 7   Tutoring           2392 non-null   int64  
 8   ParentalSupport    2392 non-null   int64  
 9   Extracurricular    2392 non-null   int64  
 10  Sports             2392 non-null   int64  
 11  Music              2392 non-null   int64  
 12  Volunteering       2392 non-null   int64  
 13  GPA                2392 non-null   float64
 14  GradeClass         2392 non-null   float64
dtypes: float64(3), int64(12)
memory usage: 280.4 KB


## **Tratamiento del conjunto de datos:**

Se realiza la carga del Dataframe (df) para visualizar y entender la estructura de los datos y su interpretabilidad. Se evidencia que la mayoría de las variables se encuentran codificadas, por lo anterior, se realizó un proceso de homologación de variables categóricas reemplazando los valores numéricos codificados por sus respectivas etiquetas descriptivas.

Esta transformación permite mejorar la interpretabilidad del conjunto de datos, facilitar el análisis estadístico y asegurar una lectura más clara de los resultados en los análisis exploratorios y visualizaciones, manteniendo al mismo tiempo la equivalencia de la información original.

Las variables a homologar son las siguientes:

| Variable | Homologación |
|---|---|
| Gender | 0 → Masculino<br>1 → Femenino |
| Ethnicity | 0 → Caucasian<br>1 → African American<br>2 → Asian<br>3 → Other |
| ParentalEducation | 0 → None<br>1 → High School<br>2 → Some College<br>3 → Bachelor's<br>4 → Higher |
| Tutoring | 0 → No<br>1 → Sí |
| ParentalSupport | 0 → None<br>1 → Low<br>2 → Moderate<br>3 → High<br>4 → Very High |
| Extracurricular | 0 → No<br>1 → Sí |
| Sports | 0 → No<br>1 → Sí |
| Music | 0 → No<br>1 → Sí |
| Volunteering | 0 → No<br>1 → Sí |
| GradeClass | 0 → A<br>1 → B<br>2 → C<br>3 → D<br>4 → F |

Además, se evidencia con la función info que no hay datos nulos, duplicados, por lo tanto no se requiere hacer limpieza o ajuste adicional al df.

In [6]:
# Se crea nuevo dataframe para no alterar el original y se reemplazan los valores numéricos por sus correspondientes categorías para facilitar la interpretación de los datos.
datos = df.copy()

# 1. Gender
datos["Gender"] = datos["Gender"].replace({
    0: "Male",
    1: "Female"
})

# 2. Ethnicity
datos["Ethnicity"] = datos["Ethnicity"].replace({
    0: "Caucasian",
    1: "African American",
    2: "Asian",
    3: "Other"
})

# 3. ParentalEducation
datos["ParentalEducation"] = datos["ParentalEducation"].replace({
    0: "None",
    1: "High School",
    2: "Some College",
    3: "Bachelor's",
    4: "Higher"
})

# 4. Tutoring
datos["Tutoring"] = datos["Tutoring"].replace({
    0: "No",
    1: "Yes"
})

# 5. ParentalSupport
datos["ParentalSupport"] = datos["ParentalSupport"].replace({
    0: "None",
    1: "Low",
    2: "Moderate",
    3: "High",
    4: "Very High"
})

# 6. Extracurricular
datos["Extracurricular"] = datos["Extracurricular"].replace({
    0: "No",
    1: "Yes"
})

# 7. Sports
datos["Sports"] = datos["Sports"].replace({
    0: "No",
    1: "Yes"
})

# 8. Music
datos["Music"] = datos["Music"].replace({
    0: "No",
    1: "Yes"
})

# 9. Volunteering
datos["Volunteering"] = datos["Volunteering"].replace({
    0: "No",
    1: "Yes"
})

# 10. GradeClass
datos["GradeClass"] = datos["GradeClass"].replace({
    0: "A",
    1: "B",
    2: "C",
    3: "D",
    4: "F"
})

# Se verifica el resultado
datos.head(10)

,StudentID,Age,Gender,Ethnicity,ParentalEducation,StudyTimeWeekly,Absences,Tutoring,ParentalSupport,Extracurricular,Sports,Music,Volunteering,GPA,GradeClass
0,1001,17,Female,Caucasian,Some College,19.833723,7,Yes,Moderate,No,No,Yes,No,2.929196,C
1,1002,18,Male,Caucasian,High School,15.408756,0,No,Low,No,No,No,No,3.042915,B
2,1003,15,Male,Asian,Bachelor's,4.210570,26,No,Moderate,No,No,No,No,0.112602,F
3,1004,17,Female,Caucasian,Bachelor's,10.028829,14,No,High,Yes,No,No,No,2.054218,D
4,1005,17,Female,Caucasian,Some College,4.672495,17,Yes,High,No,No,No,No,1.288061,F
5,1006,18,Male,Caucasian,High School,8.191219,0,No,Low,Yes,No,No,No,3.084184,B
6,1007,15,Male,African American,High School,15.601680,10,No,High,No,Yes,No,No,2.748237,C
7,1008,15,Female,African American,Higher,15.424496,22,Yes,Low,Yes,No,No,No,1.360143,F
8,1009,17,Male,Caucasian,None,4.562008,1,No,Moderate,No,Yes,No,Yes,2.896819,C
9,1010,16,Female,Caucasian,High School,18.444466,0,No,High,Yes,No,No,No,3.573474,A


## **Análisis Descriptivo Univariado:**

**Observación:** En el conjunto de datos se encuentran variables tanto Categóricas como Cuantitativas, clasificadas en los siguientes tipos:

---

### **Binarias**
Son variables categóricas que solo tienen dos posibles valores.

- `Gender`: Representa el género del estudiante. Valores:
  - `0`: Masculino
  - `1`: Femenino

- `Tutoring`: Indica si el estudiante recibe tutorías académicas. Valores:
  - `0`: No
  - `1`: Sí

- `Extracurricular`: Indica si el estudiante participa en actividades extracurriculares. Valores:
  - `0`: No
  - `1`: Sí

- `Sports`: Indica si el estudiante participa en actividades deportivas. Valores:
  - `0`: No
  - `1`: Sí

- `Music`: Indica si el estudiante participa en actividades musicales. Valores:
  - `0`: No
  - `1`: Sí

- `Volunteering`: Indica si el estudiante participa en actividades de voluntariado. Valores:
  - `0`: No
  - `1`: Sí

---

### **Continuas**
Son variables numéricas que pueden tomar cualquier valor dentro de un rango continuo.

- `StudyTimeWeekly`: Cantidad de horas de estudio semanal registradas por el estudiante.  
  - Representa el tiempo dedicado al estudio fuera de clase.

- `GPA`: Promedio académico del estudiante.  
  - Se mide en una escala aproximada de 0 a 4 puntos, donde valores más altos indican mejor rendimiento académico.

---

### **Discretas**
Son variables numéricas que solo pueden tomar valores enteros contables.

- `Age`: Edad del estudiante.
  - Aproximadamente entre 15 y 18 años.

- `Absences`: Número de ausencias escolares registradas.
  - Valores enteros mayores o iguales a 0.

---

### **Nominales**
Son variables categóricas sin orden lógico entre sus categorías.

- `Ethnicity`: Grupo étnico del estudiante. Categorías:
  - `0`: Caucasian
  - `1`: African American
  - `2`: Asian
  - `3`: Other

- `StudentID`: Identificador único del estudiante.
  - Valores numéricos únicos utilizados únicamente para identificar cada registro en el conjunto de datos.

---

### **Ordinales**
Son variables categóricas que presentan niveles con un orden lógico o jerárquico.

- `ParentalEducation`: Nivel educativo de los padres del estudiante. Valores:
  - `0`: None
  - `1`: High School
  - `2`: Some College
  - `3`: Bachelor's
  - `4`: Higher

- `ParentalSupport`: Nivel de apoyo parental percibido por el estudiante. Valores:
  - `0`: None
  - `1`: Low
  - `2`: Moderate
  - `3`: High
  - `4`: Very High

- `GradeClass`: Clasificación del rendimiento académico del estudiante según su GPA. Categorías:
  - `0`: A (GPA ≥ 3.5)
  - `1`: B (3.0 ≤ GPA < 3.5)
  - `2`: C (2.5 ≤ GPA < 3.0)
  - `3`: D (2.0 ≤ GPA < 2.5)
  - `4`: F (GPA < 2.0)

## **Feature Engineering**

Se realiza el feature engineering, creando una nueva variable significativa: Total activities.

**Total_activities**

Esta variable agrupa la participación en actividades extracurriculares (columnas Extracurricular, Sports, Music, Volunteering).Lo que nos permitirá evaluar si la participación en actividades adicionales influye positiva o negativamente en el desempeño académico.


In [7]:
datos_fe = datos.copy()

datos_fe["Total_activities"] = (
    (datos_fe["Extracurricular"] == "Yes").astype(int) +
    (datos_fe["Sports"] == "Yes").astype(int) +
    (datos_fe["Music"] == "Yes").astype(int) +
    (datos_fe["Volunteering"] == "Yes").astype(int)
)

datos_fe.head()


,StudentID,Age,Gender,Ethnicity,ParentalEducation,StudyTimeWeekly,Absences,Tutoring,ParentalSupport,Extracurricular,Sports,Music,Volunteering,GPA,GradeClass,Total_activities
0,1001,17,Female,Caucasian,Some College,19.833723,7,Yes,Moderate,No,No,Yes,No,2.929196,C,1
1,1002,18,Male,Caucasian,High School,15.408756,0,No,Low,No,No,No,No,3.042915,B,0
2,1003,15,Male,Asian,Bachelor's,4.210570,26,No,Moderate,No,No,No,No,0.112602,F,0
3,1004,17,Female,Caucasian,Bachelor's,10.028829,14,No,High,Yes,No,No,No,2.054218,D,1
4,1005,17,Female,Caucasian,Some College,4.672495,17,Yes,High,No,No,No,No,1.288061,F,0


In [8]:
datos_fe["Total_activities"].value_counts().sort_index()

Total_activities
0     679
1    1065
2     527
3     113
4       8
Name: count, dtype: int64

In [9]:
datos_fe.groupby("Total_activities")["GPA"].mean().round(2)

Total_activities
0    1.75
1    1.90
2    2.09
3    1.98
4    2.08
Name: GPA, dtype: float64

**Análisis variable Total_activities**  

La mayoría de los estudiantes participa en una o ninguna actividad extracurricular, siendo muy pocos los que tienen una alta carga de actividades.

En cuanto al rendimiento académico, se observa que el GPA tiende a mejorar a medida que aumenta la participación en actividades, pasando de 1.75 en estudiantes sin actividades a valores cercanos o superiores a 2.0 en quienes participan en dos o más.

Sin embargo, en niveles muy altos de participación no se evidencia una mejora significativa adicional, lo que sugiere que un nivel moderado de actividades podría ser más beneficioso.

En conclusión, participar en actividades extracurriculares parece tener un impacto positivo en el rendimiento académico, especialmente cuando se mantiene un equilibrio.


## **Análisis Descriptivo**

In [10]:
# Se hace una descripción estadística de las variables numéricas utilizando el método '.describe()' para obtener medidas como la media, desviación estándar, valores mínimos y máximos, etc.

print("Variables continuas")
display(datos_fe[["StudyTimeWeekly","GPA"]].describe().round(2))
print("----------------------------------------------------------------------")

print("Variables discretas")
display(datos_fe[["Age","Absences","Total_activities"]].describe().round(2))
print("----------------------------------------------------------------------")

print("Variables binarias")
display(datos_fe[["Gender","Tutoring","Extracurricular","Sports","Music","Volunteering"]].describe())

Variables continuas


,StudyTimeWeekly,GPA
count,2392.00,2392.00
mean,9.77,1.91
std,5.65,0.92
min,0.00,0.00
25%,5.04,1.17
50%,9.71,1.89
75%,14.41,2.62
max,19.98,4.00


----------------------------------------------------------------------
Variables discretas


,Age,Absences,Total_activities
count,2392.00,2392.00,2392.00
mean,16.47,14.54,1.04
std,1.12,8.47,0.85
min,15.00,0.00,0.00
25%,15.00,7.00,0.00
50%,16.00,15.00,1.00
75%,17.00,22.00,2.00
max,18.00,29.00,4.00


----------------------------------------------------------------------
Variables binarias


,Gender,Tutoring,Extracurricular,Sports,Music,Volunteering
count,2392,2392,2392,2392,2392,2392
unique,2,2,2,2,2,2
top,Female,No,No,No,No,No
freq,1222,1671,1475,1666,1921,2016


En las variables continuas, se observa que el tiempo de estudio semanal y el GPA presentan cierta variabilidad entre estudiantes, lo que indica diferencias en hábitos de estudio y rendimiento académico.

En las variables discretas, la edad muestra poca variación (grupo relativamente homogéneo), mientras que las ausencias y actividades tienen valores más separados entre sí (mayor desviación estándar y rango).

En cuanto a las variables binarias, predominan las categorías negativas (“No”), indicando baja participación en actividades extracurriculares y tutorías.

Se procede a realizar una tabla dinámica, en la que se analiza el promedio académico (GPA) cruzando dos variables: la cantidad de actividades extracurriculares (Total_activities) y el nivel de apoyo de los padres (ParentalSupport).

In [11]:
pivot_gpa = pd.pivot_table(
    datos_fe,
    values="GPA",
    index="Total_activities",
    columns="ParentalSupport",
    aggfunc="mean"
).round(2)

orden = ["None", "Low", "Moderate", "High", "Very High"]
pivot_gpa = pivot_gpa.reindex(columns=orden)

display(pivot_gpa)

ParentalSupport,None,Low,Moderate,High,Very High
Total_activities,,,,,
0,1.38,1.61,1.73,1.90,2.05
1,1.64,1.71,1.85,2.03,2.25
2,1.61,2.02,2.11,2.21,2.31
3,1.51,1.80,1.94,2.27,2.04
4,NaN,2.44,2.42,2.14,1.31


El apoyo de los padres parece ser un factor determinante en el rendimiento del estudiante, independientemente de cuántas actividades realice. Se puede observar que hay una tendencia clara: a mayor apoyo, mejor promedio académico.

Los niveles de apoyo parental "Very High" y "High" suelen tener los promedios más altos (superando casi siempre el 2.0). Por otro lado, los niveles "None" y "Low" presentan los promedios más bajos. Sin embargo, como hallazgo curioso se evidencia que los estudiantes con apoyo "Low" o "Moderate" que realizan 4 actividades tienen promedios bastante altos, 2.44 y 2.42, respectivamente, superando incluso a los de apoyo High.

## **Ranking por actividades extra**

In [19]:
# Reestructuramos el DataFrame de formato ancho a formato largo, con el fin de que todas las actividades estén en una sola columna 

actividades = ["Extracurricular", "Sports", "Music", "Volunteering"]

long_acts = datos_fe.melt(
    id_vars="GPA",
    value_vars=actividades,
    var_name="Actividad",
    value_name="Participa"
)

# Sacamos el GPA promedio solo para quienes participan

impacto_actividad = (
    long_acts[long_acts["Participa"] == "Yes"]
    .groupby("Actividad")["GPA"]
    .mean()
    .reset_index()
)

# Realizamos el ranking de actividades

impacto_actividad["Ranking"] = (
    impacto_actividad["GPA"]
    .rank(ascending=False, method="dense")
)

impacto_actividad.sort_values("Ranking").reset_index(drop=True)


,Actividad,GPA,Ranking
0,Music,2.041664,1.0
1,Extracurricular,2.015357,2.0
2,Sports,1.986381,3.0
3,Volunteering,1.913089,4.0


El ranking evidencia que, entre las actividades analizadas, la participación en Music se asocia con el mayor GPA promedio, seguida por Extracurricular y Sports, mientras que Volunteering presenta el GPA promedio más bajo. Esto sugiere que las actividades con mayor componente estructurado o cognitivo tienden a relacionarse con un mejor desempeño académico, aunque todas contribuyen en distinta medida al rendimiento estudiantil.

## **Segmentación por género de estudiantes con mejor promedio académico**

In [13]:
# Filtramos estudiantes con GradeClass A y B (rango de mejor GPA)

df_seg = datos_fe[datos_fe["GradeClass"].isin(["A", "B"])].copy()

# Definimos las condiciones y segmentos que queremos para nuestra segmentación

conditions = [
    df_seg["Gender"] == "Female",
    df_seg["Gender"] == "Male"
]

choices = [
    "Mujeres sobresalientes (A) o buen desempeño (B)",
    "Hombres sobresalientes (A) o buen desempeño (B)"
]

# Creamos la variable de segmentación

df_seg["Segmento_estudiante"] = np.select(
    conditions,
    choices,
    default="Otros alto desempeño"
)

# Resumen

segment_summary = (
    df_seg
    .groupby("Segmento_estudiante")
    .agg(
        estudiantes=("StudentID", "count"),
        GPA_promedio=("GPA", "mean"),
        estudio_promedio=("StudyTimeWeekly", "mean"),
        ausencias_promedio=("Absences", "mean")
    )
    .sort_values("GPA_promedio", ascending=False)
)

display(segment_summary.round(2))


,estudiantes,GPA_promedio,estudio_promedio,ausencias_promedio
Segmento_estudiante,,,,
Mujeres sobresalientes (A) o buen desempeño (B),186,3.09,11.41,4.79
Hombres sobresalientes (A) o buen desempeño (B),190,2.97,11.25,6.07


Las mujeres con calificación A o B presentan un GPA promedio ligeramente superior y menos ausencias en comparación con los hombres del mismo nivel académico, a pesar de que ambos grupos dedican un tiempo de estudio semanal muy similar. Esto sugiere que, dentro del alto desempeño académico, existen diferencias de género asociadas principalmente a la asistencia y no al esfuerzo de estudio, lo cual puede influir en el rendimiento final.

## **Perfil académico y hábitos de estudio de estudiantes con alto desempeño y participación extracurricular**

In [14]:
actividades = ["Extracurricular", "Sports", "Music", "Volunteering"]

#  Buscamos la actividad que más practican

actividad_mas_practicada = (
    datos_fe[actividades]
    .eq("Yes")
    .sum()
    .idxmax()
)

actividad_mas_practicada

'Extracurricular'

In [15]:
datos_fe[actividad_mas_practicada].value_counts()

Extracurricular
No     1475
Yes     917
Name: count, dtype: int64

In [16]:
# Filtramos con query (Mujeres, GradeClass A o B, Participan en la actividad más practicada)

casos_interesantes = datos_fe[
    (datos_fe["Gender"] == "Female") &
    (datos_fe[actividad_mas_practicada] == "Yes") &
    (datos_fe["GradeClass"].isin(["A", "B"]))
][[
    "StudentID", "Age", "Gender", "GradeClass",
    "GPA", "StudyTimeWeekly", "Absences",
    actividad_mas_practicada
]]

display(casos_interesantes.head(10))

,StudentID,Age,Gender,GradeClass,GPA,StudyTimeWeekly,Absences,Extracurricular
9,1010,16,Female,A,3.573474,18.444466,0,Yes
38,1039,15,Female,B,3.018906,2.949078,3,Yes
44,1045,18,Female,A,4.000000,18.921512,1,Yes
91,1092,18,Female,A,3.506920,7.724728,0,Yes
126,1127,16,Female,A,3.535004,5.494101,1,Yes
140,1141,18,Female,B,3.139482,0.785839,2,Yes
146,1147,17,Female,A,3.588888,9.647124,1,Yes
154,1155,15,Female,B,3.075619,4.304876,2,Yes
183,1184,17,Female,B,3.273999,17.507265,9,Yes
196,1197,16,Female,B,3.117354,1.989925,0,Yes


In [17]:
# Filtramos con query (Hombres, GradeClass A o B, Participan en la actividad más practicada)

casos_interesantes = datos_fe[
    (datos_fe["Gender"] == "Male") &
    (datos_fe[actividad_mas_practicada] == "Yes") &
    (datos_fe["GradeClass"].isin(["A", "B"]))
][[
    "StudentID", "Age", "Gender", "GradeClass",
    "GPA", "StudyTimeWeekly", "Absences",
    actividad_mas_practicada
]]

display(casos_interesantes.head(10))

,StudentID,Age,Gender,GradeClass,GPA,StudyTimeWeekly,Absences,Extracurricular
5,1006,18,Male,B,3.084184,8.191219,0,Yes
65,1066,16,Male,B,3.261855,18.111643,6,Yes
119,1120,15,Male,A,3.619461,19.565309,3,Yes
124,1125,15,Male,B,3.498257,2.580728,1,Yes
134,1135,17,Male,B,3.161436,19.337311,5,Yes
160,1161,17,Male,B,3.375994,0.995620,4,Yes
179,1180,17,Male,B,3.274233,15.209292,1,Yes
212,1213,15,Male,B,3.258295,5.725181,1,Yes
284,1285,16,Male,B,3.270262,9.589100,7,Yes
399,1400,15,Male,B,3.035694,3.371981,7,Yes


Esta segmentación refleja que la participación extracurricular, combinada con buenos hábitos académicos generales, se asocia con un alto rendimiento, aunque las estrategias y niveles de esfuerzo varían significativamente entre los estudiantes, tanto para hombres como mujeres.

A pesar de este buen rendimiento (GradeClass A y B), se observa una alta variabilidad en los hábitos de estudio: mientras algunas estudiantes dedican más de 18 horas semanales, otras presentan tiempos de estudio considerablemente bajos (menos de 3 horas) y aun así mantienen GPA superiores a 3.0. Esto sugiere que el rendimiento no depende únicamente del tiempo de estudio, sino también de otros factores como la eficiencia del aprendizaje o el contexto académico.

En cuanto a la asistencia, también se observa que estudiantes con altas ausencias pueden tener un buen desempeño.

## **Top**

In [18]:
top_combinations = (
    datos_fe
    .groupby(
        ["Gender", "GradeClass", "ParentalSupport", "Total_activities"],
        observed=False
    )
    .agg(
        estudiantes=("StudentID", "count"),
        GPA_promedio=("GPA", "mean"),
        ausencias_promedio=("Absences", "mean"),
        estudio_promedio=("StudyTimeWeekly", "mean")
    )
    .reset_index()
)

# 

top_combinations = top_combinations[
    top_combinations["estudiantes"] >= 15
]

#


top_combinations = top_combinations.sort_values(
    ["GPA_promedio", "estudiantes"],
    ascending=[False, False]
)

display(top_combinations.head(10).round(3))


,Gender,GradeClass,ParentalSupport,Total_activities,estudiantes,GPA_promedio,ausencias_promedio,estudio_promedio
107,Male,B,High,1,23,3.240,3.391,12.298
23,Female,B,Moderate,1,25,3.060,4.280,11.488
15,Female,B,High,1,16,3.012,5.625,11.875
116,Male,B,Moderate,1,21,2.911,6.381,12.742
126,Male,C,High,0,18,2.755,6.167,10.656
36,Female,C,Low,1,20,2.718,5.850,10.286
136,Male,C,Moderate,1,29,2.691,5.897,9.539
40,Female,C,Moderate,1,28,2.690,6.786,10.305
127,Male,C,High,1,23,2.672,8.652,9.620
137,Male,C,Moderate,2,16,2.644,9.000,12.267


El top de combinaciones muestra que los estudiantes con calificación B y niveles moderados o altos de apoyo parental tienen los GPA promedio más altos, especialmente cuando participan en al menos una actividad extracurricular. A medida que disminuye la calificación (nivel C), se observa una tendencia de reducción del GPA, acompañada de mayores ausencias y, en general, menor tiempo de estudio, lo que indica que el apoyo parental y la participación extracurricular pueden ser factores que ayudan a mejorar el rendimiento académico.